### DISCUSSION TITLE
How to run channel just on input file change

### DISCUSSION THREAD

**Discussion by tGitm** [nextgenhealthcare/connect#5068] [Q&A] [answered]
Hi 👋 
I would like to run my channels just when the input file change. For example I have a channel with file reader and a destination with database writer. On channel start, I read input HL7 file and import it's data to my database and this runs every x seconds. Now I want to run channel just when the input file change.

This is my channel:
[MESI - WEDA - PatientImport.zip](https://github.com/nextgenhealthcare/connect/files/8167469/MESI.-.WEDA.-.PatientImport.zip)

**Comment by pacmano1** [comment, marked as answer]
Mirth doesn't get file change event notifications unless you write a channel to create an event which in turn mirth can act upon.

**Reply by tGitm** [reply]
Ok, thanks. In my case I have a channel which writes patient data to database, but when I start it, the channel keeps importing data.Now I would like to change that, so channel only writes to database if there is new HL7 file in folder.

**Reply by tGitm** [reply]
I did a destination javascript writer, which deletes file after it was read in mirth.

**Reply by franklwambo** [reply]
> I did a destination javascript writer, which deletes file after it was read in mirth.

You might want to archive the read files to a given folder instead of deleting say for a month or so. So many things happen during data processing, and you might want to review original files occasionally till your channel is stable.
